cluster.go

```diff
@@ -217,7 +217,7 @@ func (c *Cluster) handleSerfEvent(e serf.Event) {
 
 	case serf.EventMemberFailed:
 		c.logger.Debug("Handling unresponsive member event")
-		go c.updateEventMembers(e)
+		go c.removeEventMembers(e)
```

Comment:
I _think_ this is a "feature".  AFAICT, when serf fails to connect to a host it emits `EventMemberFailed` and updates the host's state in an internal cache.  It then periodically "reaps" hosts that remain unresponsive for some amount of time and emits `EventMemberReap` events (at which point we remove them from the ring)

Fail:
https://github.com/hashicorp/serf/blob/768a3c43b2e2b7bb18a70a41f75cbf8d75f506f5/serf/serf.go#L888-L935

Reap:
https://github.com/hashicorp/serf/blob/768a3c43b2e2b7bb18a70a41f75cbf8d75f506f5/serf/serf.go#L1369-L1400

I'm not necessarily against adopting this behavior, but you might try setting the reap interval to something very small and see if it behaves as I'm describing:
https://github.com/hashicorp/serf/blob/master/serf/config.go#L101